tests/test/scheduler/test_remote_mpi_worlds.cpp

```diff
@@ -366,4 +362,187 @@ TEST_CASE_METHOD(RemoteCollectiveTestFixture,
     senderThread.join();
     localWorld.destroy();
 }
+
+TEST_CASE_METHOD(RemoteMpiTestFixture,
+                 "Test sending sync and async message to same host",
```

Comment:
This is the test that exposed the flaws in the thread pool (and would make the current master fail).